File: lab_1/Vector.cs
namespace lab_1;

public class Vector
{
    private double x;
    private double y;

    public Vector(double xCoord, double yCoord)
    {
        x = xCoord;
        y = yCoord;
    }

    public double X
    {
        get { return x; }
    }

    public double Y
    {
        get { return y; }
    }

    public double CalculateRadius()
    {
        return Math.Sqrt(x * x + y * y);
    }

    public double CalculateAngle()
    {
        return Math.Atan2(y, x);
    }
}

File: lab_1_plus/Vector.cpp
#include "Vector.h"
#include <cmath>

Vector::Vector(double xCoord, double yCoord) {
    x = xCoord;
    y = yCoord;
}

double Vector::getX() const {
    return x;
}

double Vector::getY() const {
    return y;
}

double Vector::calculateRadius() const {
    return std::sqrt(x*x + y*y);
}

double Vector::calculateAngle() const {
    return std::atan2(y, x);
}



